cellprofiler/modules/identifylinearobjects.py

```diff
@@ -0,0 +1,561 @@
+            same linear object.
+            <p>Select <i>%(YES)s</i> to have <b>IdentifyLinearObjects</b>
+            automatically calculate the distance from the linear object length
+            and width. Select <i>%(NO)s</i> to set the distances manually.</p>"""%globals())
+
+        self.space_distance = cps.Float(
+            "Spatial distance", 5, minval = 1,doc = """
+            <i>(Used only if not automatically calculating distance parameters)</i><br>
+            Enter the distance for calculating the linear object centers, in units of pixels.
+            The linear object centers must be at least many pixels apart for the centers to
+            be considered two separate linear objects.""")
+
+        self.angular_distance = cps.Float(
+            "Angular distance", 30, minval = 1,doc = """
+            <i>(Used only if automatically calculating distance parameters)</i><br>
+            <b>IdentifyLinearObjects</b> calculates the linear object centers at different
+            angles. Two linear object centers are considered to represent different
+            linear objects if their angular distance is larger than this number. The
+            number is measured in degrees.""")
+        
+        self.overlap_within_angle = cps.Binary(
+            "Combine if overlapping and within angular distance?", False,doc = """
+            This setting determines whether or not
+            <b>IdentifyLinearObjects</b> merges putative linear objects that are within the
+            angular distance specified AND in which pixels from the two 
+            linear objects overlap.
+            <p>Select <i>%(YES)s</i> to have <b>IdentifyLinearObjects</b>
+            merge these linear objects. Select <i>%(NO)s</i> use only the spatial
+            distance and angular distance parameters above.</p>"""%globals())
+
+    def settings(self):
+        '''The settings as they appear in the pipeline file'''
+        return [self.image_name, self.object_name, self.object_width,
+                self.object_length, self.angle_count, self.wants_automatic_distance,
+                self.space_distance, self.angular_distance, self.overlap_within_angle]
+
+    def visible_settings(self):
+        '''The settings as they appear in the user interface'''
+        result = [self.image_name, self.object_name, self.object_width,
+                  self.object_length, self.angle_count, self.wants_automatic_distance]
+        if not self.wants_automatic_distance:
+            result += [self.space_distance, self.angular_distance, self.overlap_within_angle]
+        return result
+
+    def run(self, workspace):
+        '''Run the algorithm on one image set'''
+        #
+        # Get the image as a binary image
+        #
+        image_set = workspace.image_set
+        image = image_set.get_image(self.image_name.value,
+                                    must_be_binary = True)
+        mask = image.pixel_data
```

Comment:
`mask` is a confusing name. This looks more like image data than an image mask. Maybe `data` is more appropriate.